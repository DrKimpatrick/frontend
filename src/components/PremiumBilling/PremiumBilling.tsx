import React, { FC } from 'react';
import { withRouter } from 'react-router-dom';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import ArrowRightAltTwoToneIcon from '@material-ui/icons/ArrowRightAltTwoTone';
import CheckIcon from '@material-ui/icons/Check';
import NavBar from '../Layout/NavBar/NavBar';
import MainBackground from '../Layout/MainBackground/MainBackground';

type props = {};
const PremiumBilling: FC<props> = () => {
  return (
    <>
      <NavBar />
      <section className="premium-billing-section w-1/3 m-auto text-textGray">
        <div className="flex relative h-auto my-8">
          <div className="back-arrow cursor-pointer">
            <ArrowBackTwoToneIcon />
          </div>
          <h1 className="font-bold text-xl title">Premium Billing Options</h1>
        </div>

        <div className="text-textGray mt-8  card py-1">
          <div className="font-bold py-3 px-16 check-icon">
            <CheckIcon />
          </div>
          <div className="py-3 card-content">
            <h3 className="font-bold text-textGray card-title">
              Top Talent Listing
            </h3>
            <div className="text-textGray mt-1 card-description">
              Employers will see your profile before standard & basic members.
            </div>
          </div>
        </div>

        <div className="text-textGray mt-3 card py-1">
          <div className="font-bold py-3 px-16 check-icon">
            <CheckIcon />
          </div>
          <div className="py-3 card-content">
            <h3 className="font-bold text-textGray card-title">
              Employment Verification
            </h3>
            <div className="text-textGray mt-1 card-description">
              Your employment records will be pre verified expediting the hiring
              process.
            </div>
          </div>
        </div>

        <div className="text-textGray mt-3 card py-1">
          <div className="font-bold py-3 px-16 check-icon">
            <CheckIcon />
          </div>
          <div className="py-3 card-content">
            <h3 className="font-bold text-textGray card-title">
              Education Verification
            </h3>
            <div className="text-textGray mt-1 card-description">
              Your education records will be pre verified expediting the hiring
              process.
            </div>
          </div>
        </div>

        <div className="text-textGray mt-3 card py-1">
          <div className="font-bold py-3 px-16 check-icon">
            <CheckIcon />
          </div>
          <div className="py-3 card-content">
            <h3 className="font-bold text-textGray card-title">
              3 Skill Certifications Vouchers
            </h3>
            <div className="text-textGray mt-1 card-description">
              3 Vouchers for any tech talent skill certification test.
            </div>
          </div>
        </div>

        <div className="text-textGray mt-3 card py-1">
          <div className="font-bold py-3 px-16 check-icon">
            <CheckIcon />
          </div>
          <div className="py-3 card-content">
            <h3 className="font-bold text-textGray card-title">
              15% off Certification & Training
            </h3>
            <div className="text-textGray mt-1 card-description">
              15% off all tech talent certifications & training.
            </div>
          </div>
        </div>

        <div className="w-3/4 mx-auto my-8 bottom-card">
          <div className="flex justify-between w-full">
            <div className="pricing ">
              <span>Monthly</span>
              <button
                className="monthly w-full text-white py-2 mt-1 flex justify-center"
                type="button"
              >
                <span>$19.99</span>
              </button>
            </div>
            <div className="pricing ">
              <span>Annual 15% Discount</span>
              <button
                className="annually w-full text-white py-2 mt-1 flex justify-center"
                type="button"
              >
                <span>$203.90</span>
              </button>
            </div>
          </div>

          <input
            type="text"
            placeholder="Promo Code"
            className="w-full py-2 mt-3 flex justify-center promo-code"
          />
        </div>

        <div className="flex justify-center mt-12">
          <button
            data-testid="next-button"
            className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
            type="button"
          >
            <span className="">Next</span> <ArrowRightAltTwoToneIcon />
          </button>
        </div>
      </section>
      <MainBackground />
    </>
  );
};

export default withRouter(PremiumBilling);
