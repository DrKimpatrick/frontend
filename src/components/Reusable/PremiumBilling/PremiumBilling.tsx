import React from 'react';
import { useHistory, useLocation, withRouter } from 'react-router-dom';
import ArrowBackTwoToneIcon from '@material-ui/icons/ArrowBackTwoTone';
import InfoOutlinedIcon from '@material-ui/icons/InfoOutlined';
import CheckIcon from '@material-ui/icons/Check';
import CloseIcon from '@material-ui/icons/Close';
import NavBar from '../Layout/NavBar/NavBar';
import MainBackground from '../Layout/MainBackground/MainBackground';
import './styles.scss';

const PremiumBilling = () => {
  const {
    state: { plans: productPlans, features }
  } = useLocation<{ plans: any[]; features: any[] }>();
  const history = useHistory();

  return (
    <>
      <NavBar />
      <section className="premium-billing-section w-1/3 m-auto text-gray-texts">
        <div className="flex relative h-auto my-8">
          <div className="back-arrow cursor-pointer">
            <ArrowBackTwoToneIcon onClick={() => history.goBack()} />
          </div>
          <h1 className="font-bold text-xl title">Premium Billing Options</h1>
        </div>

        {features.map((feature: any, i: number) => (
          <div key={i} className="text-gray-texts mt-8 card py-1">
            <div className="font-bold py-3 px-16 check-icon">
              {feature.available ? (
                <CheckIcon className="text-green-700" />
              ) : (
                <CloseIcon className="text-red-700" />
              )}
            </div>
            <div className="py-3 card-content">
              <h3
                className={`-ml-1 font-bold ${
                  feature.available ? 'text-green-700' : 'text-red-700'
                } card-title`}
              >
                {feature.name}
              </h3>
              <div className="text-gray-texts mt-1 card-description">
                <InfoOutlinedIcon className="-ml-8 mr-2" fontSize="small" />
                {feature.detail}
              </div>
            </div>
          </div>
        ))}

        <div className="w-3/4 mx-auto my-8 bottom-card">
          <div className="flex justify-between w-full">
            <div className="pricing ">
              <span>Monthly</span>
              <button
                className="monthly w-full text-white py-2 mt-1 flex justify-center"
                type="button"
                onClick={() =>
                  history.push({
                    pathname: '/payment',
                    state: {
                      plan: productPlans[1],
                      featureChoice: 'premium'
                    }
                  })
                }
              >
                <span>{productPlans[1].amount}</span>
              </button>
            </div>
            <div className="pricing ">
              <span>Annual 15% Discount</span>
              <button
                className="annually w-full text-white py-2 mt-1 flex justify-center"
                type="button"
                onClick={() =>
                  history.push({
                    pathname: '/payment',
                    state: {
                      plan: productPlans[0],
                      featureChoice: 'premium'
                    }
                  })
                }
              >
                <span>{productPlans[0].amount}</span>
              </button>
            </div>
          </div>

          <input
            type="text"
            placeholder="Promo Code"
            className="w-full py-2 mt-3 flex justify-center promo-code"
          />
        </div>
      </section>
      <MainBackground />
    </>
  );
};

export default withRouter(PremiumBilling);
