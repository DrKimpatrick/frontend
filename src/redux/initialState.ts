import users from 'redux/initialStates/users';

const initialState = {
  users
};

export default initialState;
