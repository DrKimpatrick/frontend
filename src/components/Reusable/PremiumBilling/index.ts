import PremiumBilling from './PremiumBilling';

export { PremiumBilling };
