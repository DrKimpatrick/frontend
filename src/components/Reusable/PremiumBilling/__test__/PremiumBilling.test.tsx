import { cleanup } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { BrowserRouter as Router } from 'react-router-dom';
import PremiumBilling from '../PremiumBilling';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import initialState from '../../../../redux/initialState';

let store: any;

const middleware = [thunk];

const mockStore = configureMockStore(middleware);

const mockHistoryPush = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('`PremiumBilling` component', () => {
  beforeEach(() => {
    store = mockStore(initialState);
  });

  afterEach(cleanup);

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDom.render(
      <Provider store={store}>
        <Router>
          <PremiumBilling />
        </Router>
      </Provider>,
      div
    );
  });

  it('matches the snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <PremiumBilling />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
