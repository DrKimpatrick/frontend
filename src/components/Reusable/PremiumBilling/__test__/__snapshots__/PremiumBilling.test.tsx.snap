// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`PremiumBilling\` component matches the snapshot 1`] = `
Array [
  <nav
    className="flex items-center bg-white-800 px-20 py-5 flex-wrap"
  >
    <a
      className="text-gray-800"
      href="/"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="logo w-24"
        src="logo-image.png"
      />
    </a>
    <div
      className="hidden top-navbar w-full lg:inline-flex lg:flex-grow lg:w-auto"
      id="navigation"
    >
      <div
        className="lg:inline-flex lg:flex-row lg:ml-auto lg:w-auto w-full lg:items-center items-start flex flex-col lg:h-auto"
      >
        <div>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              How it works
            </span>
          </a>
          <a
            className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
            href="/"
            onClick={[Function]}
          >
            <span>
              About us
            </span>
          </a>
        </div>
        <a
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded text-gray-600 items-center justify-center hover:text-gray-700"
          href="/Login"
          onClick={[Function]}
        >
          <span>
            Login
          </span>
        </a>
        <button
          className="lg:inline-flex lg:w-auto w-full px-8 py-2 rounded items-center justify-center bg-getstartColor hover:bg-blue-800 text-white"
          onClick={[Function]}
          type="button"
        >
          <span>
            Get started
          </span>
        </button>
      </div>
    </div>
  </nav>,
  <section
    className="premium-billing-section w-1/3 m-auto text-textGray"
  >
    <div
      className="flex relative h-auto my-8"
    >
      <div
        className="back-arrow cursor-pointer"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
          />
        </svg>
      </div>
      <h1
        className="font-bold text-xl title"
      >
        Premium Billing Options
      </h1>
    </div>
    <div
      className="text-textGray mt-8  card py-1"
    >
      <div
        className="font-bold py-3 px-16 check-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="py-3 card-content"
      >
        <h3
          className="font-bold text-textGray card-title"
        >
          Top Talent Listing
        </h3>
        <div
          className="text-textGray mt-1 card-description"
        >
          Employers will see your profile before standard & basic members.
        </div>
      </div>
    </div>
    <div
      className="text-textGray mt-3 card py-1"
    >
      <div
        className="font-bold py-3 px-16 check-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="py-3 card-content"
      >
        <h3
          className="font-bold text-textGray card-title"
        >
          Employment Verification
        </h3>
        <div
          className="text-textGray mt-1 card-description"
        >
          Your employment records will be pre verified expediting the hiring process.
        </div>
      </div>
    </div>
    <div
      className="text-textGray mt-3 card py-1"
    >
      <div
        className="font-bold py-3 px-16 check-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="py-3 card-content"
      >
        <h3
          className="font-bold text-textGray card-title"
        >
          Education Verification
        </h3>
        <div
          className="text-textGray mt-1 card-description"
        >
          Your education records will be pre verified expediting the hiring process.
        </div>
      </div>
    </div>
    <div
      className="text-textGray mt-3 card py-1"
    >
      <div
        className="font-bold py-3 px-16 check-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="py-3 card-content"
      >
        <h3
          className="font-bold text-textGray card-title"
        >
          3 Skill Certifications Vouchers
        </h3>
        <div
          className="text-textGray mt-1 card-description"
        >
          3 Vouchers for any tech talent skill certification test.
        </div>
      </div>
    </div>
    <div
      className="text-textGray mt-3 card py-1"
    >
      <div
        className="font-bold py-3 px-16 check-icon"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
      </div>
      <div
        className="py-3 card-content"
      >
        <h3
          className="font-bold text-textGray card-title"
        >
          15% off Certification & Training
        </h3>
        <div
          className="text-textGray mt-1 card-description"
        >
          15% off all tech talent certifications & training.
        </div>
      </div>
    </div>
    <div
      className="w-3/4 mx-auto my-8 bottom-card"
    >
      <div
        className="flex justify-between w-full"
      >
        <div
          className="pricing "
        >
          <span>
            Monthly
          </span>
          <button
            className="monthly w-full text-white py-2 mt-1 flex justify-center"
            type="button"
          >
            <span>
              $19.99
            </span>
          </button>
        </div>
        <div
          className="pricing "
        >
          <span>
            Annual 15% Discount
          </span>
          <button
            className="annually w-full text-white py-2 mt-1 flex justify-center"
            type="button"
          >
            <span>
              $203.90
            </span>
          </button>
        </div>
      </div>
      <input
        className="w-full py-2 mt-3 flex justify-center promo-code"
        placeholder="Promo Code"
        type="text"
      />
    </div>
    <div
      className="flex justify-center mt-12"
    >
      <button
        className="next-btn text-white hover:bg-gray-800 font-semibold py-1 px-3 w-32 rounded-sm shadow flex justify-around"
        data-testid="next-button"
        type="button"
      >
        <span
          className=""
        >
          Next
        </span>
         
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.01 11H4v2h12.01v3L20 12l-3.99-4v3z"
          />
        </svg>
      </button>
    </div>
  </section>,
  <div
    className="mainBackground"
  >
    <img
      alt=""
      src="motif-image.png"
      style={
        Object {
          "backgroundColor": "none",
        }
      }
    />
  </div>,
]
`;
